src/v/pandaproxy/test/kafka_client_cache.cc

```diff
@@ -27,24 +30,40 @@ using namespace std::chrono_literals;
 namespace pandaproxy {
 class test_client_cache : public kafka_client_cache {
 public:
-    explicit test_client_cache(size_t max_size)
+    explicit test_client_cache(
+      size_t max_size, ss::timer<ss::lowres_clock>& clean_and_evict_timer)
       : kafka_client_cache(
         to_yaml(kafka::client::configuration{}, config::redact_secrets::no),
         max_size,
-        1000ms) {}
+        1000ms,
+        clean_and_evict_timer) {}
+};
+
+struct cache_wrapper {
+    ss::timer<ss::lowres_clock> clean_and_evict_timer;
+    ss::gate g;
+    test_client_cache client_cache;
+
+    cache_wrapper(size_t max_size)
+      : client_cache{max_size, std::reference_wrapper(clean_and_evict_timer)} {
+        clean_and_evict_timer.set_callback([this] {
+            ssx::spawn_with_gate(
+              g, [this] { return client_cache.clean_stale_clients(); });
+        });
+    }
```

Comment:
nit: I'd have been tempted to implement this as part of `test_client_cache` to avoid the code churn